Migrate 'Matrix-based security' from wiki to jenkins.io

megathaum opened the pull request:
PR for https://github.com/jenkins-infra/jenkins.io/issues/3307

Migrated [Matrix-based security](https://wiki.jenkins.io/display/JENKINS//Matrix-based+security) from wiki to jenkins.io under `jenkins.io/content/doc/book/system-administration/security/matrix-based-security.adoc`.

- Replace 'slave' with 'agent'
- Remove notes regarding deprecated ('dangerous') permissions
- Replaced link to wiki in [Security Jenkins](https://www.jenkins.io/doc/book/system-administration/security/) to the new page 
- Added additional explanation about matrix-based security settings from Managing Jenkins > [Managing Security](https://www.jenkins.io/doc/book/managing/security/)

Created issue https://github.com/jenkins-infra/jenkins.io/issues/3348 to further merge the 2 pages on the topic of securing Jenkins: 

* System Administration > [Security Jenkins](https://www.jenkins.io/doc/book/system-administration/security/)
* Managing Jenkins > [Managing Security](https://www.jenkins.io/doc/book/managing/security/)

MarkEWaite commented on content/doc/book/system-administration/security.adoc:
```diff
@@ -27,6 +27,8 @@ Securing Jenkins has two aspects to it.
 
 == Access Control
 
+=== Access control for users
+
```
I don't gain much from a heading that is immediately followed by a heading with no intervening content.  Could we just rely on the parent heading, then let the user understand that "Access control for builds" is a sub-heading of "Access Control"?
```suggestion
```

megathaum replied:
I will remove it, just thought it would look "asymmetric" and perhaps odd in the TOC.

MarkEWaite commented on content/doc/book/system-administration/security/matrix-based-security.adoc:
```diff
@@ -0,0 +1,133 @@
+---
+title: Matrix Based Security
```
This page needs an entry added to the `_chapter.yml` file so that it will appear in the contents list on the left.  Without that entry, it is difficult for the reader to find the page, since it is only accessible through the hyperlink in the security.adoc file.

megathaum replied:
This page is in a subfolder `security`, together with `build_authorization.adoc` and `configuring-content-security-policy.adoc`. Adding it to `_ chapter.yml` place them all at the same level. See screenshot below. Is this ok, or can we create another level in the table of contents? 

<img width="282" alt="Screenshot 2020-06-11 at 21 18 49" src="https://user-images.githubusercontent.com/6340362/84429933-43d6ab00-ac29-11ea-8b43-bae61db2034a.png">

MarkEWaite replied:
I think that they should be at the same level in the table of contents.  I'd prefer it be placed immediately below "Securing Jenkins" so that the two related pages are adjacent to one another in the table of contents, but am also OK if they are not adjacent.

megathaum replied:
No problem. I'll make the changes.

megathaum commented on content/doc/book/system-administration/security/matrix-based-security.adoc:
```diff
@@ -0,0 +1,133 @@
+Matrix-based security is one of the authorization strategies available
+for https://www.jenkins.io/doc/book/system-administration/security/[securing
+Jenkins]. It allows you to grant specific permissions to users and
+groups. 
```
Sure, I will remove the trailing spaces.

megathaum commented:
Review changes pushed:

- trailing white space
- added TOC entry in `_chapter.yaml`
- removed the intermediate section header

I have also done some additional edits:

- remove the unnecessary capital in the title
- format the document as one sentence per line as Daniel remarked previously for security related documents
- removed the wiki generated anchor IDs as we don't need them

kmartens27 commented:
As this pull request has not had any updates since June 2020, and has been marked `stalled` (as discussed in Jenkins Documentation SIG office hours)[https://docs.google.com/document/d/1ygRZnVtoIvuEKpwNeF_oVRVCV5NKcZD1_HMtWlUZguo/edit#heading=h.cao5dku7jwfn], it will be closed and linked to the [original issue](https://github.com/jenkins-infra/jenkins.io/issues/3307) that will give others an opportunity to continue the work already started here.
